docs/index.md

```diff
@@ -45,3 +45,156 @@ or the WCS API (which will also install `wcs-lib`):
 opam install wcs-api
 ```
 
+# Example
+
+In order to illustrate the use of wcs-ocaml, we are going to program a
+bot that tells a knock knock joke.
+
+Let's start to create a dialog node that says `"Knock knock"`:
+
+```ocaml
+let knock =
+  Wcs.dialog_node "Knock"
+    ~conditions: "true"
+    ~text: "Knock knock"
+	()
+```
+
+The function
+[`Wcs.dialog_node`](https://ibm.github.io/wcs-ocaml/wcs-lib/Wcs/index.html#val-dialog_node)
+creates a value of type
+[`Wcs_t.dialog_node`](https://ibm.github.io/wcs-ocaml/wcs-lib/Wcs_t/index.html#type-dialog_node)
+that corresponds to a JSON object of type
+[`DialogNode`](https://www.ibm.com/watson/developercloud/conversation/api/v1/)
+in WCS.
+
+The user is expected to ask _who is there?_. To capture this intent
+but not match exactly on this input, we can define a WCS intent with some examples to train the Natural Language Understanding component of WCS:
```

Comment:
To capture this intent without looking for an exact match we can define a WCS intent using multiple examples to train the NLU